### PR TITLE
Ftr: add binary & text encode

### PR DESCRIPTION
**Pull request by Lvnszn**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/mysql/rows.go [hunk 1]
```diff
@@ -114,6 +114,45 @@ func (row *Row) GetColumnValue(column string) (interface{}, error) {
...
+						continue
+					}
+				default:
+					continue
+				}
+			} else {
+				continue
+			}
+		}
+	}
```
Comment (jjeffcaii) [inline pkg/mysql/rows.go:145]: Need to refactor this block, too many 'continue'.

Comment (Lvnszn) [reply pkg/mysql/rows.go]: address

File: pkg/mysql/rows.go [hunk 2]
```diff
@@ -114,6 +114,38 @@ func (row *Row) GetColumnValue(column string) (interface{}, error) {
 	return nil, nil
 }
 
+func (rows *TextRow) Encode(row []*proto.Value, fields []proto.Field, columnNames []string) proto.Row {
+	var val []byte
```
Comment (AlexStocks) [inline pkg/mysql/rows.go:118]: var (
)
改成酱紫

File: pkg/mysql/rows.go [hunk 3]
```diff
@@ -114,6 +114,38 @@ func (row *Row) GetColumnValue(column string) (interface{}, error) {
...
+	var isNull bool
+	var err error
+
+	for i := 0; i < len(fields); i++ {
+		field := fields[i].(*Field)
+		val = append(val, appendLengthEncodedInteger(row[i].Raw, uint64(row[i].Len))...)
+		if err == nil {
```
Comment (AlexStocks) [inline pkg/mysql/rows.go:125]: 为啥写这行代码，我没看懂

Comment (Lvnszn) [reply pkg/mysql/rows.go]: addressed, 删掉了。

File: pkg/mysql/rows.go [hunk 4]
```diff
@@ -114,6 +114,38 @@ func (row *Row) GetColumnValue(column string) (interface{}, error) {
...
+			if !isNull {
+				switch field.fieldType {
+				case mysql.FieldTypeTimestamp, mysql.FieldTypeDateTime,
+					mysql.FieldTypeDate, mysql.FieldTypeNewDate:
+					data, err := appendDateTime(row[i].Raw, row[i].Val.(time.Time))
```
[the remark this answers is not in the document]
Comment (Lvnszn) [reply pkg/mysql/rows.go]: addressed

File: pkg/mysql/rows.go (hunk 2 above)
Comment (Lvnszn) [reply pkg/mysql/rows.go]: addressed

File: pkg/mysql/rows.go [hunk 5]
```diff
@@ -114,6 +114,41 @@ func (row *Row) GetColumnValue(column string) (interface{}, error) {
...
+					if err != nil {
+						return nil
+					}
+					val = append(val, data...)
+					if err == nil {
+						continue
+					}
+				}
+			}
+		}
```
[the remark this answers is not in the document]
Comment (Lvnszn) [reply pkg/mysql/rows.go]: 这个没了，木有 err 和 isNull 了

**Review by jjeffcaii — APPROVED** [review]

**Pull request merged**